### PR TITLE
Fix jsonDocIterator by implementing JsonIter workaround

### PR DESCRIPTION
**Pull request by CanBerker**

### CODE REVIEW

**Comment by CanBerker** [comment]
@ghislainfourny The new solution is up

**Comment by CanBerker** [comment]
@ghislainfourny Could you take a look at this ?

**Review by ghislainfourny — APPROVED** [review]
Wonderful, thank you!

File: src/main/java/org/rumbledb/utils/JsonIterUtils.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+     * If this call is made first, then all consequent inputstream or string input usages are successful.
+     *
+     * However, if a string input is parsed first and dynamic code generation of streaming inputs is triggered later.
+     * JsonIter reaches an invalid state where no streaming input can be processed correctly.
+     * This function triggers a streaming input and hence dynamic code generation.
+     * This workaround should be applied during startup for reliable use of JsonIter
+     */
+    public static void applyJsonIterFaultyInitializationWorkAround() {
+        InputStream is = new ByteArrayInputStream("{}".getBytes());
+        @SuppressWarnings("unused")
```
Comment (ghislainfourny) [inline src/main/java/org/rumbledb/utils/JsonIterUtils.java:43]: I added this to remove the warning.

**Pull request merged**